crates/parcel/src/requests/target_request.rs

```diff
@@ -0,0 +1,1107 @@
+use std::ffi::OsStr;
+use std::hash::Hash;
+use std::path::Path;
+use std::path::PathBuf;
+
+use anyhow::anyhow;
+use package_json::BrowserField;
+use package_json::BrowsersList;
+use package_json::BuiltInTargetDescriptor;
+use package_json::ModuleFormat;
+use package_json::PackageJson;
+use package_json::SourceMapField;
+use package_json::TargetDescriptor;
+use parcel_core::config_loader::ConfigLoader;
+use parcel_core::types::engines::Engines;
+use parcel_core::types::BuildMode;
+use parcel_core::types::Entry;
+use parcel_core::types::Environment;
+use parcel_core::types::EnvironmentContext;
+use parcel_core::types::OutputFormat;
+use parcel_core::types::ParcelOptions;
+use parcel_core::types::SourceType;
+use parcel_core::types::Target;
+use parcel_core::types::TargetSourceMapOptions;
+use parcel_resolver::IncludeNodeModules;
+
+use crate::request_tracker::Request;
+use crate::request_tracker::RequestResult;
+use crate::request_tracker::RunRequestContext;
+use crate::request_tracker::RunRequestError;
+
+mod package_json;
+
+pub struct TargetRequest {
```

Comment:
can we write down what is target request

Reply:
Sure